src/components/views/dialogs/keybackup/CreateKeyBackupDialog.js

```diff
@@ -98,61 +101,215 @@ export default React.createClass({
         this.props.onFinished(true);
     },
 
-    _generateKey: async function() {
+    _onOptOutClick: function() {
+        this.setState({phase: PHASE_OPTOUT_CONFIRM});
+    },
+
+    _onSetUpClick: function() {
+        this.setState({phase: PHASE_PASSPHRASE});
+    },
+
+    _onSkipPassPhraseClick: async function() {
+        this._keyBackupInfo = await MatrixClientPeg.get().prepareKeyBackupVersion();
+        this.setState({
+            copied: false,
+            phase: PHASE_SHOWKEY,
+        });
+    },
+
+    _onPassPhraseNextClick: function() {
+        this.setState({phase: PHASE_PASSPHRASE_CONFIRM});
+    },
+
+    _onPassPhraseKeyPress: function(e) {
+        if (e.key === 'Enter') {
+            this._onPassPhraseNextClick();
+        }
+    },
+
+    _onPassPhraseConfirmNextClick: async function() {
+        this._keyBackupInfo = await MatrixClientPeg.get().prepareKeyBackupVersion(this.state.passPhrase);
+        this.setState({
+            copied: false,
+            phase: PHASE_SHOWKEY,
+        });
+    },
+
+    _onPassPhraseConfirmKeyPress: function(e) {
+        if (e.key === 'Enter') {
+            this._onPassPhraseConfirmNextClick();
```

Comment:
all the keydown handlers look to give the user a way to submit even when the "next" button is disabled. Should probably have a condition either here or in the click handler to check for that condition.

Reply:
ooh, good catch